packages/expo-speech/ios/EXSpeechModule/EXSpeech.m

```diff
@@ -0,0 +1,153 @@
+// Copyright 2015-present 650 Industries. All rights reserved.
+
+#import <EXSpeechModule/EXSpeech.h>
+#import <AVFoundation/AVFoundation.h>
+#import <EXCore/EXEventEmitterService.h>
+
+@interface EXSpeechUtteranceWithId : AVSpeechUtterance
+
+@property (nonatomic) NSString* utteranceId;
+
+- (instancetype)initWithString:(NSString *)string utteranceId:(NSString *)utteranceId;
+
+@end
+
+@implementation EXSpeechUtteranceWithId
+
+- (instancetype)initWithString:(NSString *)string utteranceId:(NSString *)utteranceId
+{
+  self = [super initWithString:string];
+  if (self) {
+    _utteranceId = utteranceId;
+  }
+  return self;
+}
+
+@end
+
+@interface EXSpeech () <AVSpeechSynthesizerDelegate>
+
+@property (nonatomic, strong) AVSpeechSynthesizer *synthesizer;
+@property (nonatomic, weak) EXModuleRegistry *moduleRegistry;
+
+@end
+
+@implementation EXSpeech
+
+EX_EXPORT_MODULE(ExponentSpeech)
+
+- (void)setModuleRegistry:(EXModuleRegistry *)moduleRegistry
+{
+  _moduleRegistry = moduleRegistry;
+}
+
+- (NSArray<NSString *> *)supportedEvents
+{
+  return @[@"Exponent.speakingStarted", @"Exponent.speakingDone", @"Exponent.speakingStopped", @"Exponent.speakingError"];
+}
+
+- (void)startObserving {
+}
+
+
+- (void)stopObserving {
+}
+
+
+- (void)speechSynthesizer:(AVSpeechSynthesizer *)synthesizer
+  didStartSpeechUtterance:(AVSpeechUtterance *)utterance
+{
+  id<EXEventEmitterService> emiter = [_moduleRegistry getModuleImplementingProtocol:@protocol(EXEventEmitterService)];
+  if (emiter != nil) {
```

Comment:
In ObjC calling methods on `nil` is _safe_, so we don't need to `if` this, but it's ok to leave it as far as I'm concerned